### PR TITLE
libaec: upgrade for conan v2

### PR DESCRIPTION
**Pull request by paulharris**
A relatively simple upgrade (I hope - lets see the reviews),
This is used as an option for HDF5.

### CODE REVIEW

**Comment by paulharris** [comment]
QUESTION: The upstream libaec provides cmake files that allow libaec to pose as szip,
ie it generates SZIP_LIBRARIES, SZIP_FOUND, etc.

We don't want to support that in Conan, right?

File: recipes/libaec/all/conanfile.py [hunk 1]
```diff
@@ -21,87 +25,79 @@ class LibaecConan(ConanFile):
...
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
```
Comment (uilianries) [inline recipes/libaec/all/conanfile.py:52]:
```suggestion
```

File: recipes/libaec/all/conanfile.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir, replace_in_file
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
```
Comment (uilianries) [inline recipes/libaec/all/conanfile.py:6]:
```suggestion
from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
```
Not used

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by uilianries** [comment]
> QUESTION: The upstream libaec provides cmake files that allow libaec to pose as szip, ie it generates SZIP_LIBRARIES, SZIP_FOUND, etc.
> 
> We don't want to support that in Conan, right?

No.

**Review by jwillikers — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**